### PR TITLE
Cleanup IM message def

### PR DESCRIPTION
**Pull request by yunhanw-google**
Summary of Changes:
--Remove extra header check in all im message def headers.
--Run im schema check only under CheckSchemaValidity in debug linux/mac build
--Remove redundant schema check

### CODE REVIEW

**Review by mspang — APPROVED** [review]

**Review by msandstedt — APPROVED** [review]

**Review by yufengwangca — APPROVED** [review]

File: src/app/MessageDef/MessageDefHelper.h [hunk 1]
```diff
@@ -23,37 +23,15 @@
...
 #include <stdio.h>
 
-#ifndef CHIP_CONFIG_IM_ENABLE_SCHEMA_CHECK
-#define CHIP_CONFIG_IM_ENABLE_SCHEMA_CHECK 1
-#endif // CHIP_CONFIG_IM_ENABLE_SCHEMA_CHECK
-
 namespace chip {
 namespace app {
 
-#if CHIP_DETAIL_LOGGING
```
Comment (LuDuda) [inline src/app/MessageDef/MessageDefHelper.h:56]: Just double checking.. Is this change intentional? Or should be more like `CHIP_CONFIG_IM_ENABLE_SCHEMA_CHECK && CHIP_DETAIL_LOGGING`

**Review by LuDuda — APPROVED** [review]
